@here/harp-vectortile-datasource/lib/OmvDebugLabelsTile.ts

```diff
@@ -121,109 +139,146 @@ export class OmvDebugLabelsTile extends Tile {
         const centerZ = this.center.z;
         const pointScale = this.mapView.pixelToWorld;
         const worldOffsetX = this.computeWorldOffsetX();
+        const zHeight = 10;
 
         let pointLabelIndex = 0;
 
-        if (this.textElementGroups.count() > 0) {
-            const bluePointGeometry = new THREE.BufferGeometry();
-            const bluePointIndices = new Array<number>();
-            const bluePointPositions = new Array<number>();
+        const addPoint = (
+            xPos: number,
+            yPos: number,
+            zPos: number,
+            size: number,
+            debugGeometry: DebugGeometry
+        ) => {
+            debugGeometry.positions.push(xPos, yPos - size, zPos);
+            debugGeometry.positions.push(xPos + size, yPos, zPos);
+            debugGeometry.positions.push(xPos, yPos + size, zPos);
+            debugGeometry.positions.push(xPos - size, yPos, zPos);
+
+            const index = debugGeometry.positions.length / 3;
+
+            debugGeometry.indices.push(index - 4);
+            debugGeometry.indices.push(index - 3);
+            debugGeometry.indices.push(index - 2);
+            debugGeometry.indices.push(index - 4);
+            debugGeometry.indices.push(index - 2);
+            debugGeometry.indices.push(index - 1);
+        };
```

Comment:
better move this to a free function to make this method easier shorter an easier to follow.

Reply:
done